### ISSUE TITLE
Can't build project

### ISSUE THREAD

**Issue by baitcode** [NeoBirth/NeoBirth#11]
Hi! i'm not really good with embedded systems and may be missing something silly. I've been following  the instructions in README.md

Here is my rust version:
```
>> rustup show

Default host: x86_64-apple-darwin
rustup home:  /<redacted>/.rustup

installed targets for active toolchain
--------------------------------------

thumbv7em-none-eabihf
x86_64-apple-darwin

active toolchain
----------------

stable-x86_64-apple-darwin (default)
rustc 1.39.0 (4560ea788 2019-11-04)
```

I receive the following error when building:
```
NeoBirth git:(master) ✗ CARGO_INCREMENTAL=0 cargo build --release
   Compiling neobirth v0.1.1 (<redacted>/NeoBirth)
error: linking with `rust-lld` failed: exit code: 1
  |
  = note: "rust-lld" "-flavor" "gnu" "-L" "<redacted>.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/thumbv7em-none-eabihf/lib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/neobirth-85bc28062404c9f2.neobirth.8qroh1u1-cgu.0.rcgu.o" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/neobirth-85bc28062404c9f2.neobirth.8qroh1u1-cgu.1.rcgu.o" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/neobirth-85bc28062404c9f2.neobirth.8qroh1u1-cgu.10.rcgu.o" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/neobirth-85bc28062404c9f2.neobirth.8qroh1u1-cgu.11.rcgu.o" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/neobirth-85bc28062404c9f2.neobirth.8qroh1u1-cgu.12.rcgu.o" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/neobirth-85bc28062404c9f2.neobirth.8qroh1u1-cgu.13.rcgu.o" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/neobirth-85bc28062404c9f2.neobirth.8qroh1u1-cgu.2.rcgu.o" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/neobirth-85bc28062404c9f2.neobirth.8qroh1u1-cgu.3.rcgu.o" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/neobirth-85bc28062404c9f2.neobirth.8qroh1u1-cgu.4.rcgu.o" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/neobirth-85bc28062404c9f2.neobirth.8qroh1u1-cgu.5.rcgu.o" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/neobirth-85bc28062404c9f2.neobirth.8qroh1u1-cgu.6.rcgu.o" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/neobirth-85bc28062404c9f2.neobirth.8qroh1u1-cgu.7.rcgu.o" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/neobirth-85bc28062404c9f2.neobirth.8qroh1u1-cgu.8.rcgu.o" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/neobirth-85bc28062404c9f2.neobirth.8qroh1u1-cgu.9.rcgu.o" "-o" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/neobirth-85bc28062404c9f2" "--gc-sections" "-L" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps" "-L" "<redacted>NeoBirth/target/release/deps" "-L" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/build/trellis_m4-544507f10490787a/out" "-L" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/build/atsamd51g19a-a77ab50ab5013446/out" "-L" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/build/cortex-m-3899a01bd13feab2/out" "-L" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/build/cortex-m-rt-3922f311551936ea/out" "-L" "<redacted>.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/thumbv7em-none-eabihf/lib" "-Bstatic" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libws2812_nop_samd51-cf27806d7b76a2bd.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libtrellis_m4-6fff940a10e9a67d.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libatsamd_hal-d50b5f3526476e44.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libpaste-4f83820e641f3268.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libbitfield-20e3524d82e10127.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libatsamd51g19a-e4fe7f8aa2496efd.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libcortex_m_rt-319b06d00566489c.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libr0-fddceff958e08d36.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libcortex_m-6370709200842ce3.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libvolatile_register-881e2632aa1d1a1f.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libvcell-2dc515d65af0b207.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libaligned-6716426d24317ce6.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libbare_metal-02324ebdc35311ef.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libkeypad-c530d2fc3b073c23.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libadxl343-dd5758d47dd1ba4c.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libbitflags-25ae76e86b6a0d63.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libembedded_hal-dae8783bdf08f0b2.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libvoid-26932d9b6fadd092.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libnb-7ffd31aff45e5176.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libaccelerometer-25dcbad0f5ae6d39.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libgeneric_array-bf040142dd9a387d.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libtypenum-ac518aa13bfe22ad.rlib" "--start-group" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libpanic_halt-4ce7dd38cb4e5b47.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libsmart_leds-9b883ec72d6db7db.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/libsmart_leds_trait-e29c33deae1733d9.rlib" "<redacted>NeoBirth/target/thumbv7em-none-eabihf/release/deps/librgb-a8edf29296374514.rlib" "<redacted>.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/thumbv7em-none-eabihf/lib/librustc_std_workspace_core-9864ecb2f3ae85f5.rlib" "<redacted>.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/thumbv7em-none-eabihf/lib/libcore-a54bcae7919f0f3f.rlib" "--end-group" "<redacted>.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/thumbv7em-none-eabihf/lib/libcompiler_builtins-a509ea259a1bcbbf.rlib" "-Tlink.x" "-Bdynamic"
  = note: rust-lld: error: no memory region specified for section '.ARM.exidx'

error: aborting due to previous error
```

Short version:
```
rust-lld: error: no memory region specified for section '.ARM.exidx'
```

I've been able to find a thread on that topic here:
https://github.com/rust-embedded/cortex-m-rt/issues/157

But it's quite hard to follow as I'm not an expert in the area. I'll continue my research, but would appreciate any help at the meantime. Thank you!

**Comment by tarcieri** [comment]
This is a related issue: https://github.com/atsamd-rs/atsamd/pull/167/files

I think the `trellis_m4` crate needs to be bumped. Otherwise try `cargo update` and see if that fixes it.

**Comment by baitcode** [comment]
Thank you very much!
I was able to solve the issue by changing the dependencies section in Cargo.toml to:
```ini
[dependencies]
panic-halt = "0.2"
purezen = { version = "0.0.2", default-features = false }
trellis_m4 = { version = "0.1", features = ["adxl343", "keypad-unproven"] }
smart-leds = "0.1"
ws2812-nop-samd51 = { git = "https://github.com/smart-leds-rs/ws2812-nop-samd51.git" }
cortex-m = "0.5.8"
cortex-m-rt = "0.6.12"
cortex-m-rt-macros = "0.1.5"
```

Now I'm figuring out what is wrong with bossa. As soon as I finish, I'll update here and close the issue.
```bash
>> bossac -e -w -v -b -R -o 0x4000 --port=/dev/tty.usbmodem14101 platform dev> neobirth.bin
bossac: extra arguments found
Try 'bossac -h' or 'bossac --help' for more information
```

**Comment by baitcode** [comment]
Oh god. I'm soo inattentive - forgot to replace placeholder.
Thank you very much!

**Comment by baitcode** [comment]
For the record. you were right that this command is needed. But I left my changes in place when testing it, so i'm not sure it's sufficient.

```bash
cargo update
```
